lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -648,6 +650,17 @@ defmodule Kernel.ErrorsTest do
     assert_compile_fail CompileError,
       "nofile:1: cannot use ^x outside of match clauses",
       'x = 8; <<a, b::size(^x)>> = <<?a, ?b>>'
+
+    warning = capture_io(:stderr, fn ->
```

Comment:
I added the test on the warning because it's necessary to happen otherwise we will get green when the test is supposed to be red, but I can take it out since it's not the purpose of the test